stew/shims/tables.nim

```diff
@@ -9,4 +9,15 @@ template init*[A, B](T: type OrderedTableRef[A, B]): auto = newOrderedTable[A, B
 template init*[A](T: type CountTable[A]): auto = initCountTable[A]()
 template init*[A](T: type CountTableRef[A]): auto = newCountTable[A]()
 
+template mgetOrPutLazy*[A, B](t: Table[A, B], key: A, val: B): var B =
+  type R = B
+
+  proc setter(loc: var R): var R =
+    if loc == default(R):
```

Comment:
so if the collection had a value that was all zeroes (as opposed to not having a value), this will update it to `B` which doesn't seem to be the intent

Reply:
Yes, this version of code comes with a number of caveats, but the idea of having a shim is that it allows us to capture the intention of the user until a more general and efficient version of the API is added to the `tables` module as a template similar to `withValue`.

At the moment it's reasonable to use mostly with tables with `ref` values that don't need to store `nil`. The `default` function for the type should be cheap and not a valid value in the usage context.

Reply:
well, the that's the point a bit - my intent is not to set if the value is the default - it's to set / create only if it's not in the collection already